include/alpaka/core/Sycl.hpp

```diff
@@ -23,6 +23,26 @@
 #    include <type_traits>
 #    include <utility>
 
+// if SYCL is enabled with the AMD backend the printf will be killed because of missing compiler support
+#    ifdef __AMDGCN__
+#        include <cstdio> // the define breaks <cstdio> if it is included afterwards
+#        define printf(...)
+#    else
+
+#        ifdef __SYCL_DEVICE_ONLY__
+#            define CONSTANT __attribute__((opencl_constant))
+#        else
+#            define CONSTANT
+#        endif
+
+#        define printf(FORMAT, ...)                                                                                   \
+            do                                                                                                        \
+            {                                                                                                         \
+                static const CONSTANT char format[] = FORMAT;                                                         \
+                sycl::ext::oneapi::experimental::printf(format, ##__VA_ARGS__);                                       \
+            } while(false)
```

Comment:
This looks a bit cleaner IMO:

```suggestion
#        ifdef __SYCL_DEVICE_ONLY__
using AlpakaFormat = char const* [[clang::opencl_constant]];
#        else
using AlpakaFormat = char const*;
#        endif

#        define printf(FORMAT, ...)                                                                                   \
            do                                                                                                        \
            {                                                                                                         \
                static auto format = AlpakaFormat{FORMAT};                                   \
                sycl::ext::oneapi::experimental::printf(format, ##__VA_ARGS__);                                       \
            } while(false)
```

Reply:
ok, but it works only if I keep `const` after `static` at line 41 because SYCL cannot use a non-const static data variable in a kernel